File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        int N = nums.length;
        int[] psum = new int[N + 1];
        for(int i = 1; i <= N; ++i) {
            psum[i] = psum[i - 1] + nums[i - 1];
        }
        
        Set<Integer> set = new HashSet<>();
        for(int i = 2; i <= N; ++i) {
            set.add(psum[i - 2] % k);
            if(set.contains(psum[i] % k)) return true;
        }
        
        return false;
    }
}

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        N = len(nums)
        psum = [0] * (N + 1)
        for i in range(1, N + 1):
            psum[i] = psum[i - 1] + nums[i - 1]

        s = set()
        for i in range(2, N + 1):
            s.add(psum[i - 2] % k)
            if psum[i] % k in s:
                return True
        
        return False
        